libs/payments/ui/src/lib/client/components/SubscriptionContent/index.tsx

```diff
@@ -64,12 +78,75 @@ export const SubscriptionContent = ({
     false,
     locale
   );
+
+  async function cancelSubscriptionAtPeriodEnd() {
+    const result = await cancelSubscriptionAtPeriodEndAction(userId, subscription.id)
+    if (result.ok) {
+      setOpenDialog(true);
+      setShowCancel(false);
+      setShowCancelActionError(false);
+    } else {
+      setShowCancelActionError(true);
+    }
+  }
   return (
     <>
+      <Dialog.Root open={openDialog}>
+        <Dialog.Portal>
+          <Dialog.Overlay className='fixed inset-0 bg-black/75 z-50' />
+          <Dialog.Content
+            className='w-11/12 max-w-[545px] text-center px-7 pt-6 pb-8 rounded-xl shadow inline-block fixed top-1/2 left-1/2 -translate-x-1/2 -translate-y-1/2 z-9999 bg-white'
+            onEscapeKeyDown={() => setOpenDialog(false)}
+            onPointerDownOutside={() => setOpenDialog(false)}
+            onInteractOutside={() => setOpenDialog(false)}
+          >
+            <Dialog.Title className='font-bold leading-6 m-5'>
+              <Localized id="subscription-cancellation-dialog-title">
+                We're sorry to see you go
```

Comment:
I was going to suggest wrapping `We’re sorry to see you go` in <h2> because Localized injects translated content into a single React element, but `Dialog.Title` converts to an h2... hmm.

Reply:
```
<Dialog.Title asChild>
  <Localized id="subscription-cancellation-dialog-title">
    <h2 className="font-bold leading-6 m-5">We're sorry to see you go</h2>
  </Localized>
</Dialog.Title>
```
?

Reply:
No change :)